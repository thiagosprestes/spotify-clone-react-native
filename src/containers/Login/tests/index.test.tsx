import React from "react";
import { TouchableOpacity } from "react-native";
import renderer from "react-test-renderer";
import LoginContainer from "..";

describe("LoginContainer test", () => {
  const mockOnLogin = jest.fn();

  const containerInstance = renderer.create(
    <LoginContainer onLogin={mockOnLogin} />
  ).root;

  test("LoginContainer should be render", () => {
    const snapshotContainerInstance = renderer
      .create(<LoginContainer onLogin={mockOnLogin} />)
      .toJSON();

    expect(snapshotContainerInstance).toMatchSnapshot();
  });

  test("onLogin should be called", () => {
    containerInstance.findByType(TouchableOpacity).props.onPress();

    expect(mockOnLogin).toBeCalled();
  });
});
