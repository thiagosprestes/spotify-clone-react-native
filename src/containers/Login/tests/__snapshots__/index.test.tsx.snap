// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginContainer test LoginContainer should be render 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#121212",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/logo.png",
      }
    }
    style={
      Array [
        Object {
          "height": 75,
          "marginBottom": 50,
          "width": 250,
        },
      ]
    }
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 20,
        "height": 50,
        "justifyContent": "center",
        "opacity": 1,
        "width": 260,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "OpenSans_700Bold",
            "fontSize": 18,
          },
        ]
      }
    >
      Entrar
    </Text>
  </View>
</View>
`;
